lib/record_store/provider/oracle_cloud_dns.rb

```
@@ -0,0 +1,195 @@
+      def zone_version(zone)
+        client.get_zone(zone).data.version
+      end
+
+      # Creates a new record to the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def add(record, zone)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+        patch_add_record = [
+          OCI::Dns::Models::RecordOperation.new(
+            domain: record.fqdn,
+            rtype: record.type,
+            ttl: record.ttl,
+            rdata: record_rdata,
+            operation: 'ADD',
+          ),
+        ]
+
+        client.patch_zone_records(
+          zone,
+          OCI::Dns::Models::PatchZoneRecordsDetails.new(items: patch_add_record)
+        )
+      end
+
+      # Deletes an existing record from the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def remove(record, zone)
+        record_fqdn = Record.ensure_ends_without_dot(record.fqdn)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+        found_record = client.get_zone_records(
+          zone,
+          rtype: record.type,
+          zone_version: client.get_zone(zone).data.version,
+          domain: record_fqdn,
+        ).data.items.last
+
+        begin
+          record_hash = found_record.record_hash
+          patch_remove_record = [
+            OCI::Dns::Models::RecordOperation.new(
+              domain: record_fqdn,
+              record_hash: record_hash,
+              rtype: record.type,
+              ttl: record.ttl,
+              rdata: record_rdata,
+              operation: 'REMOVE',
+            ),
+          ]
+
+          client.patch_zone_records(
+            zone,
+            OCI::Dns::Models::PatchZoneRecordsDetails.new(items: patch_remove_record)
+          )
+        rescue NoMethodError
+          puts "An exception occurred while deleting changeset which doesn't exist"
```

Will this be considered as an error by the record store or simply ignored as we `puts` but just continue as if everything's fine ? Should we re-raise the error ?

---

For now,
- Just continue as if everything's fine
- It will be considered as an error and displaying the error message

---

Using `return unless found_record` instead of catch an error